src/plugins/discover/public/application/main/components/top_nav/esql_dataview_transition/esql_dataview_transition_modal.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FEEDBACK_LINK } from '@kbn/esql-utils';
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiText,
+  EuiCheckbox,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiLink,
+  EuiHorizontalRule,
+} from '@elastic/eui';
+
+export interface ESQLToDataViewTransitionModalProps {
+  onClose: (dismissFlag?: boolean, needsSave?: boolean) => void;
+}
+// Needed for React.lazy
+// eslint-disable-next-line import/no-default-export
+export default function ESQLToDataViewTransitionModal({
+  onClose,
+}: ESQLToDataViewTransitionModalProps) {
+  const [dismissModalChecked, setDismissModalChecked] = useState(false);
+  const onTransitionModalDismiss = useCallback((e) => {
+    setDismissModalChecked(e.target.checked);
+  }, []);
+
+  return (
+    <EuiModal
+      onClose={() => onClose()}
+      style={{ width: 700 }}
+      data-test-subj="discover-esql-to-dataview-modal"
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {i18n.translate('discover.esqlToDataviewTransitionModalTitle', {
```

Comment:
A suggestion to group the new labels under `discover.esqlToDataViewTransitionModal.<key>`:

```suggestion
          {i18n.translate('discover.esqlToDataViewTransitionModal.title', {
```

Reply:
Done https://github.com/elastic/kibana/pull/188898/commits/fcea60218ca2cb9ed2ba997763583d60e3a72729 (applied to all the others too)